### ISSUE TITLE
Bogus nodetool cfhistograms output

### ISSUE THREAD

**Issue by vladzcloudius** [scylladb/scylla#3806]
This is Scylla's bug tracker, to be used for reporting bugs only.
*Installation details*
Scylla version:  2.3
Cluster size: 1
AWS AMI: ScyllaDB 2.3.0 (ami-0fc423ac17a75570d)
Instance type: i3.2xlarge

*Description*
1) Populate the KS with the following command:
```
cassandra-stress write n=20000000 no-warmup -node <Scylla IP> -rate threads=1000 -mode native cql3 -schema keyspace="ks2" "replication(strategy=NetworkTopologyStrategy, us-east=1)"
```
(I used a c-s from the master tree of scylla-tools-java repo)
2) Restart the service: 
```
sudo systemctl stop scylla-server
sudo systemctl start scylla-server
```
3) Run the read test:
```
cassandra-stress read n=20000000 no-warmup -node <Scylla IP> -rate threads=1000 -mode native cql3 -schema keyspace="ks2"  -pop seq=1..2
```
4) Run the nodetool command:
```
$ nodetool cfhistograms ks2 standard1
ks2/standard1 histograms
Percentile  SSTables     Write Latency      Read Latency    Partition Size        Cell Count
                              (micros)          (micros)           (bytes)                  
50%             1.00              0.00              9.00        4966933177          74975550
75%             1.00              0.00             15.00        4966933177          74975550
95%             1.00              0.00             33.00        4966933177          74975550
98%             1.00              0.00            182.46        4966933177          74975550
99%             1.00              0.00            264.91        4966933177          74975550
Min             0.00              0.00              6.00        4139110982          62479626
Max             1.00              0.00          23682.00        4966933177          74975550
```

Note bogus values in the "Partition Size" and the "Cell Count" columns.

**Comment by slivne** [comment]
@amnonh - please try to map where we are extracting this info

**Comment by amnonh** [comment]
The API is mapped to:
stats_metadata.estimated_partition_size;
stats_metadata.estimated_cells_count;
In sstables/types.hh

It is updated in metadata_collector.hh:update

An example of how to check with the API directly:
curl -X GET 'http://localhost:10000/column_family/metrics/estimated_column_count_histogram/keyspace1%3Astandard1'

**Comment by slivne** [comment]
thanks @amnonh - @bhalevy - moving this to your queue

**Comment by tzach** [comment]
@bhalevy any updates?

**Comment by bhalevy** [comment]
> @bhalevy any updates?

Sorry, I just saw this.
Trying to reproduce with a dtest

**Comment by bhalevy** [comment]
Hmm, another issue with cfhistograms (I don't know if it's related)
Seen in https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-debug-random/197/testReport/nodetool_additional_test/TestNodetool/concurrent_restart_test/
Scylla version 8628ede009b339205e26b680c4e95c601fa38a6a
```
2435.3 not greater than or equal to 2619.0 : Write Latency is not monotonic: 2435.3(95% load), was 2619.0
keyspace1/standard1 histograms
Percentile  SSTables     Write Latency      Read Latency    Partition Size        Cell Count
                              (micros)          (micros)           (bytes)                  
50%             0.00           2087.00              0.00                 0                 0
75%             0.00           2619.00              0.00                 0                 0
95%             0.00           2435.30              0.00                 0                 0
98%             0.00           2631.46              0.00                 0                 0
99%             0.00           3328.86              0.00                 0                 0
Min             0.00           1583.00              0.00                 0                 0
Max             0.00          21182.00              0.00                 0                 0
```

@amnonh how can that happen?

**Comment by amnonh** [comment]
It's clearly a bug somewhere, even with restarts I don't see a reason this would happen.
How reproducible is it?

**Comment by bhalevy** [comment]
And the commit it happened includes d17b05e9112b615bdc055e12c740906aaae070d9

**Comment by slivne** [comment]
will be fixed by #6792 fix (pending review)

https://github.com/scylladb/scylla-tools-java/pull/182

**Comment by slivne** [comment]
closing #6792 is fixed
